### ISSUE TITLE
Column alias mismatch in UNION ALL chain.

### ISSUE THREAD

**Issue by StarWix** [yandex/ClickHouse#878]
Union all not work when column aliases are different in the same position. E.g:
```
SELECT 1 AS a
UNION ALL 
SELECT 1 AS b
```
`Code: 259. DB::Exception: Received from oscs60:9000, 10.10.81.60. DB::Exception: Column alias mismatch in UNION ALL chain.`

But this works:
```
SELECT 1 AS a
UNION ALL 
SELECT 1 AS a
```

**Comment by ztlpn** [comment]
Apparently you are correct and according to the standard (e.g. SQL92 item 7.10 syntax rules 12.b) UNION ALL of queries with different column aliases is perfectly legal.

**Comment by alexey-milovidov** [comment]
Fixed in #1947
